packages/test/test-end-to-end-tests/src/test/connectionMode.spec.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { MockLogger } from "@fluidframework/telemetry-utils";
+import { delay } from "@fluidframework/common-utils";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+}
+
+describeNoCompat("Connection Mode", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+    );
+
+    let mainContainer: IContainer;
+
+    const createContainer = async (logger): Promise<IContainer> => provider.createContainer(runtimeFactory, { logger });
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider();
+        const mockLogger = new MockLogger();
+        // Create a Container for the first client.
+        mainContainer = await createContainer(mockLogger);
+
+        // Set an initial key. The Container is in read-only mode so the first op it sends will get nack'd and is
+        // re-sent. Do it here so that the extra events don't mess with rest of the test.
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        mainDataStore._root.set("test", "value");
+
+        await provider.ensureSynchronized();
+    });
+
+    it("should change to read mode after timeout ", async function() {
+        if(provider.driver.type !== "odsp") {
+            this.skip();
+        }
+        const dm = mainContainer.deltaManager;
+        assert.strictEqual(dm.active, true, "connection mode should be in write");
+        await delay(1000 * 400);
```

Comment:
can you add more validation of the expected state transitions here? some ideas are quorum, audience, and connection states

Reply:
also, why do we need this test if we have the unit test?

Reply:
this one would just be more like the real scenario than the unit test

Reply:
but what additional coverage does it provide. the e2e will be more costly to develop, maintain, and run, so we should have a good reason for it if we add it

Reply:
also keep in mind if we make this test runnable in automation via mock we will not be validating the behavior of the service, as that takes too long to run.